File: BC-W4/Army/Spells/Spell/Spell.cpp
#include "Spell.h"

Spell::Spell(int actionPoints, int manaCost,const std::string& spellName) : manaCost(manaCost), actionPoints(actionPoints), spellName(spellName) {};
Spell::~Spell() {};

int Spell::getManaCost() const {
    return this->manaCost;
};

int Spell::getActionPoints() const {
    return this->actionPoints;
};

const std::string& Spell::getSpellName() const {
    return this->spellName;
};

Demon* Spell::action(Location* nLocation) {
    throw InvalidActionException();
};

File: BC-W4/ArmyJava/src/spells/Spell.java
package spells;

import exceptions.InvalidCastException;
import units.Demon;
import units.Unit;

public abstract class Spell {
    protected String spellName;
    protected int manaCost;
    protected int actionPoints;
    protected boolean isAttack;

    public Spell(String spellName, int manaCost, int actionPoints, boolean isAttack ) {
        this.spellName = spellName;
        this.manaCost = manaCost;
        this.actionPoints = actionPoints;
        this.isAttack = isAttack;
    }

    public String getSpellName() {
        return this.spellName;
    }

    public int getManaCost() {
        return this.manaCost;
    }

    public int getActionPoints() {
        return this.actionPoints;
    }

    public boolean isAttack() {
        return this.isAttack;
    }

    public void action(Unit enemy) throws InvalidCastException {
        throw new InvalidCastException();
    }
    public void heal(Unit target) throws InvalidCastException {
        throw new InvalidCastException();
    }

    public Demon action() throws InvalidCastException {
        throw new InvalidCastException();
    }

}

